features/eolearn/features/feature_manipulation.py

```diff
@@ -48,9 +49,8 @@ def __init__(
         :param filter_func: A callable that takes a numpy evaluates to bool.
         :param filter_features: A collection of features which will be filtered into a new EOPatch
         """
-        self.feature = self.parse_feature(
-            feature, allowed_feature_types=FeatureTypeSet.TEMPORAL_TYPES.difference([FeatureType.VECTOR])
-        )
+
+        self.feature = self.parse_feature(feature, allowed_feature_types=lambda fty: fty.is_temporal())
```

Comment:
Should be:

```suggestion
        self.feature = self.parse_feature(feature, allowed_feature_types=lambda fty: fty.is_temporal() and not fty.is_vector())
```

Reply:
that was my first thought as well, but if you look at L86 there is a method for filtering vector features (???)

Reply:
It seems that the feature used for filtering shouldn't be a vector, but vectors are allowed to be present under `filter_features`, where the filter is then only applied. under `_get_filtered_indices` only temporal array-type data is expected, so perhaps I'd even use `fty.is_temporal() and fty.is_array()` in this case, but the outcome is the same.

Reply:
I see that even the timestamps feature is allowed, but it's quite ugly since the variable used in ``SimpleFilterTask` is quite generic and I think often used `_get_filtered_indices` is `img`, which is then improper naming, but ultimately my final suggestion above using `is_array()` is not correct, so ignore it.

Reply:
ah i see, i completely missed that there is a feature and features :see_no_evil: 
we also need to keep in mind that this task can do filtering on timestamps, which might break once we remove it from `FeatureType`

Reverted to previous allowed types and included the suggested test